crypto/s2n_ecdsa.c

```diff
@@ -194,7 +223,8 @@ int s2n_ecdsa_pkey_matches_curve(const struct s2n_ecdsa_key *ecdsa_key, const st
     POSIX_ENSURE_REF(ecdsa_key->ec_key);
     POSIX_ENSURE_REF(curve);
 
-    int curve_id = EC_GROUP_get_curve_name(EC_KEY_get0_group(ecdsa_key->ec_key));
+    const EC_KEY *key = ecdsa_key->ec_key;
+    int curve_id = EC_GROUP_get_curve_name(EC_KEY_get0_group(key));
```

Comment:
Why is the new `key` assignment necessary here?

Reply:
its not necessary. i made this change when i had to do cast in place and left it here because I think it makes things more readable.

dont have strong preference though